OpenRA.Mods.Common/Traits/Infantry/TakeCover.cs

```diff
@@ -22,11 +24,27 @@ public class TakeCoverInfo : TurretedInfo
 		[Desc("Prone movement speed as a percentage of the normal speed.")]
 		public readonly int SpeedModifier = 50;
 
+		[Desc("Damage types that trigger prone state. Defined on the warheads.")]
+		public readonly string[] ProneTriggers = new string[0];
+
+		[FieldLoader.LoadUsing("LoadModifiers")]
+		[Desc("Damage modifiers for each damage type (defined on the warheads) that makes the unit go prone.")]
+		public readonly Dictionary<string, int> DamageModifiers = new Dictionary<string, int>();
+
 		public readonly WVec ProneOffset = new WVec(85, 0, -171);
 
 		public readonly string ProneSequencePrefix = "prone-";
 
 		public override object Create(ActorInitializer init) { return new TakeCover(init, this); }
+
+		public static object LoadModifiers(MiniYaml yaml)
+		{
+			var md = yaml.ToDictionary();
+
+			return md.ContainsKey("DamageModifiers")
+				? md["DamageModifiers"].ToDictionary(my => FieldLoader.GetValue<int>("(value)", my.Value))
+				: new Dictionary<string, int>();
```

Comment:
It would be better to default `Prone50Percent => 50` here and `TriggerProne` above so that mods that don't otherwise care don't need to explicitly specify them.  You could then drop the unnecessary upgrade rule.